commit-stream.js

```diff
@@ -1,95 +1,116 @@
-const through2  = require('through2')
-    , stripAnsi = require('strip-ansi')
+import through2 from 'through2'
+import stripAnsi from 'strip-ansi'
 
-
-module.exports = commitStream
-
-
-function commitStream (ghUser, ghProject) {
-  var commit
+export default function commitStream (ghUser, ghProject) {
+  let commit
 
   return through2.obj(onLine, onEnd)
 
   function addLine (line) {
     line = stripAnsi(line)
 
-    if (!line)
-      return
+    if (line.length === 0) return
 
-    var old, m
+    let old, m
 
     if (/^commit \w+$/.test(line)) {
       old = commit
-      commit = {
-        sha: line.split(' ')[1]
+      commit = { sha: line.split(' ')[1] }
+    } else if ((m = line.match(/^CommitDate:\s+(.*)$/)) !== null) {
+      if (commit === undefined || commit === null) {
+        throw new Error('Something unexpected occurred')
       }
-    } else if (m = line.match(/^CommitDate:\s+(.*)$/)) {
-      if (!commit)
-        throw new Error('wut?')
       commit.commitDate = m[1].trim()
-    } else if (m = line.match(/^\s*(?:Author|Co[- ]?authored[- ]?by):?\s*([^<]+) <([^>]+)>\s*$/i)) {
-      if (!commit)
-        throw new Error('wut?')
-      if (!commit.authors)
+    } else if (
+      (m = line.match(
+        /^\s*(?:Author|Co[- ]?authored[- ]?by):?\s*([^<]+) <([^>]+)>\s*$/i
+      )) !== null
+    ) {
+      if (commit === undefined || commit === null) {
+        throw new Error('Something unexpected occurred')
+      }
+      if (commit.authors === undefined || commit.authors === null) {
         commit.authors = []
+      }
       commit.authors.push({ name: m[1], email: m[2] })
-    } else if (m = line.match(/^AuthorDate:\s+(.*)$/)) {
-      if (!commit)
-        throw new Error('wut?')
-      commit.authorDate = m[1].trim()
-    } else if (m = line.match(/^Date:\s+(.*)$/)) {
-      if (!commit)
-        throw new Error('wut?')
+    } else if ((m = line.match(/^(?:AuthorDate|Date):\s+(.*)$/)) !== null) {
+      if (commit === undefined || commit === null) {
+        throw new Error('Something unexpected occurred')
+      }
       commit.authorDate = m[1].trim()
-    } else if (m = line.match(/^\s+Reviewed[- ]?By:?\s*([^<]+) <([^>]+)>\s*$/i)) {
-      if (!commit.reviewers)
+    } else if (
+      (m = line.match(/^\s+Reviewed[- ]?By:?\s*([^<]+) <([^>]+)>\s*$/i)) !==
+      null
+    ) {
+      if (commit.reviewers === undefined || commit.reviewers === null) {
         commit.reviewers = []
+      }
       commit.reviewers.push({ name: m[1], email: m[2] })
-    } else if (m = line.match(/^\s+PR(?:[- ]?URL)?:?\s*(.+)\s*$/)) {
+    } else if ((m = line.match(/^\s+PR(?:[- ]?URL)?:?\s*(.+)\s*$/)) !== null) {
```

Comment:
interesting, did standard complain about this or are you just opting to be explicit (not objecting, just intested)

Reply:
The standard tool didn't complain about it, I'm just making it explicit.